pkg/multiclustermanager/routes.go

```diff
@@ -111,9 +110,21 @@ func router(ctx context.Context, localClusterEnabled bool, tunnelAuthorizer *mcm
 	authed.PathPrefix("/v3/token").Handler(tokenAPI)
 	authed.PathPrefix("/v3").Handler(managementAPI)
 
+	// Metrics authenticated route
+	metricsAuthed := mux.NewRouter()
+	metricsAuthed.UseEncodedPath()
+	impersonatingAuth := auth.ToMiddleware(requests.NewImpersonatingAuth(sar.NewSubjectAccessReview(clusterManager)))
+	tokenReviewAuth := impersonatingAuth.Chain(auth.ToMiddleware(requests.NewTokenReviewAuth(scaledContext.K8sClient.AuthenticationV1())))
```

Comment:
Is this the right order? I'm not sure if this is correct or if the following is correct:

```go
auth.ToMiddleware(requests.NewTokenReviewAuth(scaledContext.K8sClient.AuthenticationV1())).Chain(impersonatingAuth)
```

Based on [the logic in Steve](https://github.com/rancher/steve/blob/e7119828b8f9a9ca2cb2620ba650bd0508ca463e/pkg/auth/filter.go#L47-L51) that looks like `return m(middleware(next))`, it seems like we evaluate the `middleware` provided as an argument (e.g. the tokenAuthReview) before evaluating the `m` provided as self (e.g. the impersonatingAuth), whereas what we want is for impersonation to be attempted, followed by tokenAuthReview to be checked.

Reply:
Effectively I think either way works but you're right this isn't the intended order, changed.